### PR TITLE
refactor(artifact_proxy): use Dart config

### PR DESCRIPTION
**Pull request by felangel**
## Description

- use configuration in Dart to make things strongly typed, easier to manage, and to avoid having to read/include a config.yaml in the docker image.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [X] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by eseidel — APPROVED** [review]
Lol.  I had it first in Dart then moved to yaml, now back to Dart. I think I moved to yaml mostly for the record feature.

File: packages/artifact_proxy/lib/config.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import 'package:artifact_proxy/artifact_proxy.dart';
+
+/// The proxy configuration used by the server.
+const config = ProxyConfig(
+  engineMappings: {
+    ...flutter_3_7_8,
```
Comment (bryanoltman) [inline packages/artifact_proxy/lib/config.dart:6]: We're tracking 3.7.9 in https://github.com/shorebirdtech/flutter fwiw

```
bryanoltman@boltman ~/Shorebird/_shorebird/shorebird/bin/cache/flutter (stable)
⑆ bin/flutter --version
Flutter 3.7.9 • channel stable • https://github.com/shorebirdtech/flutter.git
Framework • revision 62bd79521d (7 days ago) • 2023-03-30 10:59:36 -0700
Engine • revision ec975089ac
Tools • Dart 2.19.6 • DevTools 2.20.1
```

Comment (felangel) [reply packages/artifact_proxy/lib/config.dart:6]: sounds good we'll add 3.7.9 shortly 👍

File: packages/artifact_proxy/lib/src/artifact_proxy.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      path.contains,
+    );
+
+    final normalizedPath = shorebirdEngineRevision != null
+        ? path.replaceAll(shorebirdEngineRevision, r'$engine')
+        : path;
+
+    if (shorebirdEngineRevision == null) {
+      final location = getFlutterArtifactLocation(artifactPath: normalizedPath);
+      print('No engine revision detected, forwarding to: $location');
```
Comment (bryanoltman) [inline packages/artifact_proxy/lib/src/artifact_proxy.dart:25]: Definitely not blocking on this PR, but should we eventually move to using some kind of logger class? (probably https://pub.dev/packages/mason_logger)

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**